[BUG] Build fails with TRACE disabled

**Describe the bug**
Disable all trace in Kconfig.
```
[ 83%] Building C object CMakeFiles/sof.dir/src/drivers/dw/dma.c.o
/home/lrg/work/sof/xtensa-cnl-elf/bin/xtensa-cnl-elf-gcc  -I/home/lrg/work/sof/sof/src/platform/intel/cavs/include -I/home/lrg/work/sof/sof/src/arch/xtensa/include -I/home/lrg/work/sof/sof/src/arch/xtensa/smp/include -I/home/lrg/work/sof/sof/src/arch/xtensa/smp/xtos -I/home/lrg/work/sof/sof/src/platform/icelake/include -I/home/lrg/work/sof/sof/src/platform/icelake/include/arch -I/home/lrg/work/sof/sof/../xtensa-root/xtensa-cnl-elf/include -I/home/lrg/work/sof/sof/src/include -I/home/lrg/work/sof/sof/build_icl_gcc/generated/include  -nostdlib -fno-inline-functions -mlongcalls -O2 -g -Wall -Werror -Wl,-EL -Wmissing-prototypes -mtext-section-literals -o CMakeFiles/sof.dir/src/drivers/dw/dma.c.o   -c /home/lrg/work/sof/sof/src/drivers/dw/dma.c
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c: In function 'hda_dma_wait_for_buffer_full':
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c:263:11: error: variable 'wp' set but not used [-Werror=unused-but-set-variable]
  uint32_t wp;
           ^~
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c:262:11: error: variable 'rp' set but not used [-Werror=unused-but-set-variable]
  uint32_t rp;
           ^~
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c: In function 'hda_dma_wait_for_buffer_empty':
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c:291:11: error: variable 'wp' set but not used [-Werror=unused-but-set-variable]
  uint32_t wp;
           ^~
/home/lrg/work/sof/sof/src/drivers/intel/cavs/hda-dma.c:290:11: error: variable 'rp' set but not used [-Werror=unused-but-set-variable]
  uint32_t rp;
           ^~
```
Error trace should should write only to persistent/shared buffer when TRACE is disabled.

rimage also broken too when TRACE disabled..
```

Module Reading bootloader-icl
info: ignore .bss section for bootloader module
error: can't find section .static_log_entries in module sof-icl
info: ignore .static_log_entries section for bootloader module
error: failed to read base sections -22
info: ignore .fw_ready section for bootloader module
make[2]: *** [src/arch/xtensa/CMakeFiles/run_rimage.dir/build.make:60: src/arch/xtensa/CMakeFiles/run_rimage] Error 234
  Found 22 sections, listing valid sections......
make[1]: *** [CMakeFiles/Makefile2:1357: src/arch/xtensa/CMakeFiles/run_rimage.dir/all] Error 2
	No	Start		End		Size	Type	Name
make: *** [Makefile:133: all] Error 2
	0	0x00000000	0x00000000	0x0
	1	0xb0038000	0xb00380fb	0xfb	TEXT	.boot_entry.text
	2	0xb0038120	0xb0038140	0x20	TEXT	.boot_entry.literal
	3	0xb0038150	0xb0038809	0x6b9	TEXT	.text
	4	0xb0039000	0xb0039008	0x8	DATA	.rodata

 module: input size 2012 (0x7dc) bytes 5 sections
 module: text 2004 (0x7d4) bytes
    data 8 (0x8) bytes
    bss  0 (0x0) bytes
```

--- a/src/audio/kpb.c
+++ b/src/audio/kpb.c
@@ -795,13 +795,12 @@ static uint64_t kpb_draining_task(void *arg)
 	size_t size_to_copy;
 	bool move_buffer = false;
 	uint32_t drained = 0;
-	uint64_t time_start;
-	uint64_t time_end;
+	uint64_t time;
 	uint32_t attr = 0;
 
 	trace_kpb("kpb_draining_task(), start.");
 
-	time_start = platform_timer_get(platform_timer);
+	time = platform_timer_get(platform_timer);
 
 	while (history_depth > 0) {
 		size_to_read = (uint32_t)buff->end_addr - (uint32_t)buff->r_ptr;
@@ -836,7 +835,7 @@ static uint64_t kpb_draining_task(void *arg)
 			comp_update_buffer_produce(sink, size_to_copy);
 	}
 
-	time_end =  platform_timer_get(platform_timer);
+	time =  platform_timer_get(platform_timer) - time;
 
 	/* Draining is done. Now switch KPB to copy real time stream
 	 * to client's sink. This state is called "draining on demand"
@@ -848,8 +847,7 @@ static uint64_t kpb_draining_task(void *arg)
 
 	trace_kpb("kpb_draining_task(), done. %u drained in %d ms.",
 		   drained,
-		   (time_end - time_start)
-		   / clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1));
+		   time / clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1));
 
 	return 0;
 }

--- a/src/drivers/intel/cavs/hda-dma.c
+++ b/src/drivers/intel/cavs/hda-dma.c
@@ -259,8 +259,6 @@ static int hda_dma_wait_for_buffer_full(struct dma *dma,
 	uint64_t deadline = platform_timer_get(platform_timer) +
 		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
 		PLATFORM_HOST_DMA_TIMEOUT / 1000;
-	uint32_t rp;
-	uint32_t wp;
 
 	while (!hda_dma_is_buffer_full(dma, chan)) {
 		if (deadline < platform_timer_get(platform_timer)) {
@@ -268,12 +266,11 @@ static int hda_dma_wait_for_buffer_full(struct dma *dma,
 			if (hda_dma_is_buffer_full(dma, chan))
 				return 0;
 
-			rp = host_dma_reg_read(dma, chan->index, DGBRP);
-			wp = host_dma_reg_read(dma, chan->index, DGBWP);
-
 			trace_hddma_error("hda-dmac: %d wait for buffer full "
 					  "timeout rp 0x%x wp 0x%x",
-					  dma->plat_data.id, rp, wp);
+				dma->plat_data.id,
+				host_dma_reg_read(dma, chan->index, DGBRP),
+				host_dma_reg_read(dma, chan->index, DGBWP));
 			return -ETIME;
 		}
 	}
@@ -287,8 +284,6 @@ static int hda_dma_wait_for_buffer_empty(struct dma *dma,
 	uint64_t deadline = platform_timer_get(platform_timer) +
 		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
 		PLATFORM_HOST_DMA_TIMEOUT / 1000;
-	uint32_t rp;
-	uint32_t wp;
 
 	while (!hda_dma_is_buffer_empty(dma, chan)) {
 		if (deadline < platform_timer_get(platform_timer)) {
@@ -296,12 +291,11 @@ static int hda_dma_wait_for_buffer_empty(struct dma *dma,
 			if (hda_dma_is_buffer_empty(dma, chan))
 				return 0;
 
-			rp = host_dma_reg_read(dma, chan->index, DGBRP);
-			wp = host_dma_reg_read(dma, chan->index, DGBWP);
-
 			trace_hddma_error("hda-dmac: %d wait for buffer empty "
 					  "timeout rp 0x%x wp 0x%x",
-					  dma->plat_data.id, rp, wp);
+				dma->plat_data.id,
+				host_dma_reg_read(dma, chan->index, DGBRP),
+				host_dma_reg_read(dma, chan->index, DGBWP));
 			return -ETIME;
 		}
 	}

